doc/source/guide/in_depth/code/functional_indepth_example.py

```diff
@@ -0,0 +1,124 @@
+import odl
+
+
+# Here we define the functional
+class MyFunctional(odl.solvers.Functional):
+
+    """This is my functional: ``||x||_2^2 + <x, y>``."""
+
+    def __init__(self, space, linear_term):
+        """Initialize a new instance."""
+        # This comand calls the init of Functional and sets a number of
+        # parameters associated with a functional. All but domain have default
+        # values if not set.
+        super().__init__(space=space, linear=False, grad_lipschitz=2)
```

Comment:
> python 2 is still the most popular

I doubt that. My educated guess is that 80 % of the people go with the default of their distribution, which is 2.7 in most cases. I'm quite sure the statistics among those who actively choose is a different one.